packages/cadl-ranch/src/actions/helper.ts

```diff
@@ -0,0 +1,104 @@
+import { HttpMethod, ServiceRequestFile } from "@azure-tools/cadl-ranch-api";
+import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
+import FormData from "form-data";
+
+export interface ServiceRequest {
+  endPoint: string;
+  options?: {
+    requestBody?: any;
+    files?: ServiceRequestFile[];
+    config?: AxiosRequestConfig<any> | undefined;
+  };
+}
+
+function checkAndAddFormDataIfRequired(request: ServiceRequest) {
+  if (request.options?.config?.headers?.["Content-Type"] === "multipart/form-data") {
+    const formData = new FormData();
+    if (request.options?.requestBody) {
+      for (const key in request.options.requestBody) {
+        formData.append(key, JSON.stringify(request.options.requestBody[key]));
+      }
+    }
+    if (request.options.files) {
+      request.options.files.forEach((file) => {
+        formData.append(`${file.fieldname}`, file.buffer, {
+          filename: file.originalname,
+          contentType: file.mimetype,
+        });
+      });
+    }
+    request.options.requestBody = formData;
+    request.options.config = {
+      ...request.options.config,
+      headers: formData.getHeaders(),
+    };
+  }
+}
+
+function checkAndUpdateEndpoint(request: ServiceRequest) {
```

Comment:
Design change 3: This method has been added newly in this PR. This is added to accomodate the scenarios where the URLs are registered as `..../users/:id` and the client call has to happen  as `..../users/`'.